src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -670,7 +794,24 @@ private void EmitModule(JsonTextWriter jsonWriter, ModuleSymbol moduleSymbol, Ex
                 // If it is a template spec module, emit templateLink instead of template contents.
                 jsonWriter.WritePropertyName(moduleSemanticModel is TemplateSpecSemanticModel ? "templateLink" : "template");
                 {
-                    TemplateWriterFactory.CreateTemplateWriter(moduleSemanticModel, this.settings).Write(jsonWriter);
+                    var nestedTemplateOffset = jsonWriter.CurrentPos;
+
+                    var writer = TemplateWriterFactory.CreateTemplateWriter(moduleSemanticModel, this.settings);
+                    writer.Write(jsonWriter);
+
+                    if (writer is TemplateWriter templateWriter && moduleSemanticModel is SemanticModel moduleSemanticModelObj)
+                    {
+                        var childAbsolutePath = moduleSemanticModelObj.SourceFile.FileUri.AbsolutePath;
+                        var parentAbsolutePath = Path.GetDirectoryName(this.context.SemanticModel.SourceFile.FileUri.AbsolutePath)!;
```

Comment:
You're representing the child module relative to the parent module - but don't you need to represent the child module represent to the 'root' module?

E.g. if you've got the following structure:
* `main.bicep` has a module referencing `nested/child.bicep`
* `nested/child.bicep` has a module referencing `again/grandchild.bicep`

In your sourcemap, you will need to represent the 3 files as:
`main.bicep`, `nested/child.bicep` & `nested/again/grandchild.bicep` (note the 3rd is relative to `main.bicep`, not `nested/child.bicep`)

Reply:
Glad you caught this, I forgot to revisit this issue. I got lost in thinking about edge cases for your suggestion--like if modules were on separate drives. Let me know if you have any thoughts around this.